### PR TITLE
feat: achievement vendor and vendor feedback

### PR DESCRIPTION
**Pull request by aronwk-aaron**
Remove hardcoded achievement vendor logic to now work with any achievements vendor
Check if a player should be able to buy an item from the achievement vendor
Log cheat detection for trying to buy items that can't be bought
Better feedback when trying to buy or sell something and can't

Tested that buying from normal vendors and achievement vendors works

### CODE REVIEW

**Comment by aronwk-aaron** [comment]
Lot 13732 (TEST COMMENDATION VENDOR) now works since it's no longer hardcoded to be 13806

File: dDatabase/CDClientDatabase/CDClientTables/CDMissionsTable.cpp [hunk 1]
```diff
@@ -79,7 +79,7 @@ void CDMissionsTable::LoadValuesFromDatabase() {
 		entries.push_back(entry);
 		tableData.nextRow();
 	}
-
+	std::ranges::sort(entries, [](const CDMissions& lhs, const CDMissions& rhs) { return lhs.id < rhs.id; });
```
Comment (Xiphoseer) [inline dDatabase/CDClientDatabase/CDClientTables/CDMissionsTable.cpp:82]: Yknow, CDClient is sorted by ID if we were to start mmapping it.

File: dDatabase/CDClientDatabase/CDClientTables/CDMissionsTable.cpp [hunk 2]
```diff
@@ -118,3 +118,25 @@ const CDMissions& CDMissionsTable::GetByMissionID(uint32_t missionID, bool& foun
 	return Default;
 }
 
+const std::set<int32_t> CDMissionsTable::GetMissionsForReward(LOT lot) {
+	std::set<int32_t> toReturn {};
+	for (const auto& entry : GetEntries()) {
```
Comment (Xiphoseer) [inline dDatabase/CDClientDatabase/CDClientTables/CDMissionsTable.cpp:123]: This seems expensive

**Review by Xiphoseer — APPROVED** [review]
Lgtm

File: dDatabase/CDClientDatabase/CDClientTables/CDMissionsTable.cpp (hunk 1 above)
Comment (EmosewaMC) [reply dDatabase/CDClientDatabase/CDClientTables/CDMissionsTable.cpp:82]: was left over from a previous thing we were doing to speed up the searches.  Could be removed now mr aron

File: dDatabase/CDClientDatabase/CDClientTables/CDMissionsTable.cpp (hunk 2 above)
Comment (aronwk-aaron) [reply dDatabase/CDClientDatabase/CDClientTables/CDMissionsTable.cpp:122]: added a cache for this... it is indeed expensive, but this is also what the client does on every load of the UI for the achievement vendor, but instead we are just doing it on buy

**Review by Xiphoseer — APPROVED** [review]

**Review by jadebenn — APPROVED** [review]

**Pull request merged**